src/main/java/io/vlingo/http/resource/ResourceHandler.java

```diff
@@ -47,6 +47,14 @@
   protected ResourceHandler() {
   }
 
+  /**
+   * Construct my default state with a {@code Stage}.
+   * @param stage the Stage to set as my stage
+   */
+  protected ResourceHandler(Stage stage) {
```

Comment:
@Florian-Schoenherr This is good initiative, thanks. I have one concern, one suggestion, and one pointer. /cc @danilo-ambrosio 

- This base class is meant for use internally when a special case of `ResourceHandler` is in use. It is one that is created from configuration defined in `vlingo-http.properties`. Note that there is not only a `Stage` but also a `Context` instance variable, and both are package scope. The scope allows the server internals to set the `Stage` and `Context` on the fly, after construction has already taken place. This is because the `Context` changes with every request and response. This may normally be considered poor design, but for the purpose of the server internals it is exactly what's needed. And because it is package scope not client code can set those variables (only the server implementation). So supporting the new constructor is misleading. It means you can now use the `Stage` but the `Context` will never be set (always `null`). The reason that this base is being used is wrong by design. It should be used only for configuration resources. That's why I said the issue must be with the code generation.

- We could use a new base class that would be the base of this class. It would only support `Stage` things and could be used by dynamic resources, including those that are code generated. The problem is I don't have a good name for this because `Resource` is already used and so is `ResourceRequestHandler` (which would be wrong anyway). Perhaps `BaseResourceHandler` or `AbstractResourceHandler`, but I dislike naming things this way when the client will use them directly.

- Please always used `final` when declaring method parameters. It's a contract to the client that this method will not change the parameter passed, and to yourself because reassignment is impossible (although with Java the reassignment is benign, happening only on the stack). The point is a method should rarely, if ever, mess with parameter values.

Reply:
- having no `final` was an oversight, would change
- the ResourceHandler-javadoc says "`/** My {@code Stage} within which my backing {@code Actor} resides. */`", that's why I just said "well, even if it's dynamic, this should hold the Actor's Stage". If that should stay like it is, we can also generate this inside every Resource: 

 ```
private final Stage stage;
private final Logger logger;
public SomeResource(final Stage stage) {
        this.stage = stage;
        this.logger = stage.world().defaultLogger();
}
```
Just thought that would be redundant if the base-class has it.

- we could rename to have `ConfigurationResourceHandler` and `DynamicResourceHandler`, with base-class `ResourceHandler` or something but this would break the API everywhere.
I don't know how `ResourceHandler` is used internally, so I'm just throwing around thoughts.
So `extends ResourceHandler` is used for both config and dynamic resources? Maybe `ResourceHandler` could find out which of the both it is and then either use Server internals logic or the "pass in via ctor"-way
- > The scope allows the server internals to set the Stage and Context on the fly, after construction has already taken place.

So this can't work with dynamic resources?

Reply:
But then there's also this on `ResourceHandler`:
```
<p>NOTE: Resources that provide an independent {@code routes()}
 * implementation are not required to extend {@code ResourceHandler}
 * and are not pooled by the {@code DispatcherActor}.</p>
```
So should we try "just not extending", and have the local logger instance from my previous comment?

Reply:
@That is what I thought of previously, but thought it might be best in a base-base class. Either way, `ResourceHandler` should not be extended.